### PR TITLE
fix(CI): retry docker interactions

### PR DESCRIPTION
**Pull request by gavin-stackrox**
## Description

Docker interactions with quay.io and registry.access.redhat.com can be flaky. Pepper them with 5 retries (with backoff). 

- https://github.com/stackrox/stackrox/actions/runs/7047231619/job/19181738533#step:10:466
- https://github.com/stackrox/stackrox/actions/runs/7047438542/job/19181339078#step:10:9
- https://github.com/stackrox/stackrox/actions/runs/7048193941/job/19184691408
- https://github.com/stackrox/stackrox/actions/runs/7048199495/job/19184707722
...

## Checklist
- [x] Investigated and inspected CI test results

### Here I tell how I validated my change

CI

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -353,22 +353,22 @@ jobs:
 
       - name: Build main images
         run: |
-          GOOS=linux GOARCH=${{ matrix.arch }} make docker-build-main-image
+          GOOS=linux GOARCH=${{ matrix.arch }} scripts/lib.sh retry 5 true make docker-build-main-image
```
Comment (GrimmiMeloni) [inline .github/workflows/build.yaml:356]: Does this actually push? If not, retrying the build itself is not necessary...?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -353,22 +353,22 @@ jobs:
...
 
       - name: Check debugger presence in the main image
         run: make check-debugger
 
       - name: Build roxctl image
         run: |
-          GOOS=linux GOARCH=${{ matrix.arch }} make docker-build-roxctl-image
+          GOOS=linux GOARCH=${{ matrix.arch }} scripts/lib.sh retry 5 true make docker-build-roxctl-image
```
Comment (GrimmiMeloni) [inline .github/workflows/build.yaml:363]: same question here

**Review by GrimmiMeloni — APPROVED** [review]
This looks solid to me.
2 remarks on possible overeager retries. But could also be required - depending on what the target entails Please adjust as you see fit @gavin-stackrox

File: .github/workflows/build.yaml (hunk 1 above)
Comment (gavin-stackrox) [reply .github/workflows/build.yaml:356]: > Does this actually push? If not, retrying the build itself is not necessary...?

The flake in the build is in the `FROM <image>` part, so its the pull that I'm looking to retry. But you do bring up a good point. If someone breaks the dockerfile it'll retry 5 times with backoff before they get an error :scream: And at about ~3 minutes per build, that would be suboptimal. I'll drop to 3 retries.

**Pull request merged**